### ISSUE TITLE
`signature` failure on `Float64`

### ISSUE THREAD

**Issue by willtebbutt** [invenia/ExprTools.jl#39]
The following happens:
```julia
julia> m = which(Float64, (Float64, ))
(::Type{T})(x::T) where T<:Number
     @ Core boot.jl:792

julia> ExprTools.signature(m)
ERROR: MethodError: no method matching parameters(::TypeVar)
Closest candidates are:
  parameters(::UnionAll)
   @ ExprTools ~/ml/ad_playground/Taped.jl/dev/ExprTools/src/type_utils.jl:8
  parameters(::DataType)
   @ ExprTools ~/ml/ad_playground/Taped.jl/dev/ExprTools/src/type_utils.jl:9
  parameters(::Union)
   @ ExprTools ~/ml/ad_playground/Taped.jl/dev/ExprTools/src/type_utils.jl:10

Stacktrace:
 [1] type_parameters(sig::Type)
   @ ExprTools ~/ml/ad_playground/Taped.jl/dev/ExprTools/src/method.jl:239
 [2] type_parameters
   @ ~/ml/ad_playground/Taped.jl/dev/ExprTools/src/method.jl:232 [inlined]
 [3] signature(m::Method; extra_hygiene::Bool)
   @ ExprTools ~/ml/ad_playground/Taped.jl/dev/ExprTools/src/method.jl:47
 [4] signature(m::Method)
   @ ExprTools ~/ml/ad_playground/Taped.jl/dev/ExprTools/src/method.jl:39
 [5] top-level scope
   @ REPL[32]:1
```
Julia version 1.9.2, ExprTools version 0.1.9.

I'm trying to figure out a fix, but not sure whether I really understand what `type_parameters` is meant to be doing in this context.

**Comment by willtebbutt** [comment]
Is it as simple as adding the following method?
```julia
ExprTools.parameters(sig::TypeVar) = [sig]
```
i.e. is `ExprTools.parameters` meant to return a vector of `TypeVar`s?

**Comment by willtebbutt** [comment]
Resolved by #40
